src/vm_x86.dasc

```diff
@@ -124,16 +124,46 @@
 |.define CFRAME_SPACE,	aword*7			// Delta for esp (see <--).
 |.macro saveregs_
 |  push edi; push esi; push ebx
+|.if WIN
+|  push 0
+|  fs; push dword [0]
+|.endif
 |  sub esp, CFRAME_SPACE
 |.endmacro
 |.macro saveregs
 |  push ebp; saveregs_
 |.endmacro
 |.macro restoreregs
 |  add esp, CFRAME_SPACE
+|.if WIN
+|  fs; pop dword [0]
+|  pop edi					// Just to increment esp by 4
+|.endif
 |  pop ebx; pop esi; pop edi; pop ebp
 |.endmacro
 |
+|.if WIN
```

Comment:
This makes the stack 8 byte aligned, but not 16 byte aligned. That might crash some C function deep down in the stack, when storing SIMD registers in stack slots.

Quite a few C compilers align the stack to 16 bytes in `main()` and ensure it stays that way in the frame setup of all C functions compiled by them. Any intermediate frame that violates this assumption will cause trouble.